arrows/vxl/pixel_feature_extractor.cxx

```diff
@@ -173,49 +261,100 @@ ::filter( kwiver::vital::image_container_sptr input_image )
     if( enable_gray )
     {
       // TODO consider vil_convert_to_grey_using_rgb_weighting
-      filtered_images.push_back( vil_convert_to_grey_using_average( vxl_image ) );
+      auto vxl_gray_sptr =
+        vil_convert_to_grey_using_average(
+          vxl::image_container::vital_to_vxl( input_image->get_image() ) );
+      auto vxl_gray = vil_convert_cast( pix_t(), vxl_gray_sptr );
+      filtered_images.push_back( vxl_gray );
     }
   }
 
   if( enable_color_commonality )
   {
     // 1 channel
-    auto color_commonality = color_commonality_filter->filter( input_image );
-    filtered_images.push_back(
-        vxl::image_container::vital_to_vxl( color_commonality->get_image() ) );
+    auto color_commonality = convert_to_typed_vil_image_view< pix_t >(
+      color_commonality_filter->filter( input_image ) );
+
+    filtered_images.push_back( color_commonality );
   }
   if( enable_high_pass_box )
   {
-    auto high_pass_box = high_pass_box_filter->filter( input_image );
-    // 2 channels
-    filtered_images.push_back(
-        vxl::image_container::vital_to_vxl( high_pass_box->get_image() ) );
+    auto high_pass_box = convert_to_typed_vil_image_view< pix_t >(
+      high_pass_box_filter->filter( input_image ) );
+
+    // Legacy BurnOut models expect these channels to be incorrectly ordered
+    // TODO Remove this code when we no longer need to train models using
+    // legacy code
+    auto first_plane = vil_plane( high_pass_box, 0 );
+    auto second_plane = vil_plane( high_pass_box, 1 );
+    auto temp = vil_copy_deep( first_plane );
+    first_plane.deep_copy( second_plane );
+    second_plane.deep_copy( temp );
+
+    // 3 channels
+    filtered_images.push_back( high_pass_box );
   }
   if( enable_high_pass_bidir )
   {
-    auto high_pass_bidir = high_pass_bidir_filter->filter( input_image );
-    // 2 channels
-    filtered_images.push_back(
-        vxl::image_container::vital_to_vxl( high_pass_bidir->get_image() ) );
+    auto high_pass_bidir = convert_to_typed_vil_image_view< pix_t >(
+      high_pass_bidir_filter->filter( input_image ) );
+    // 3 channels
+    filtered_images.push_back( high_pass_bidir );
   }
+
+  vil_image_view< double > double_variance;
+  if( enable_average || enable_normalized_variance )
+  {
+    auto multiplane_variance = convert_to_typed_vil_image_view< double >( input_image );
+    if( multiplane_variance.nplanes() == 3 )
+    {
+      vil_convert_planes_to_grey( multiplane_variance, double_variance );
```

Comment:
Just double checking that this makes sense.  These three planes represent variance for R, G, and B?  Because `vil_convert_planes_to_grey` does a weighted average of three channels using weights for RGB to CIE luminance conversion.  E.g. the weight for the green channel is 10 times that of the blue channel, because green is perceptually more important than blue for humans.

Reply:
If this is what the old burnout does, it's fine to leave it as is for parity, but this might be one of those things to revisit later.